crates/fuel-core/src/service/sub_services.rs

```diff
@@ -173,7 +176,20 @@ pub fn init_sub_services(
     #[cfg(not(feature = "p2p"))]
     let p2p_adapter = P2PAdapter::new();
 
-    let update_algo = StaticAlgorithmUpdater::new(config.static_gas_price);
+    let updater_metadata = UpdaterMetadata::V0(V0Metadata {
+        new_exec_price: config.starting_gas_price,
+        min_exec_gas_price: config.min_gas_price,
+        exec_gas_price_change_percent: config.gas_price_change_percent,
+        l2_block_height: last_height.into(),
+        l2_block_fullness_threshold_percent: config.gas_price_threshold_percent,
+    });
+    // TODO: This looks wrong, why are we getting this from an adapter for some other port?
+    let block_stream = importer_adapter.events_shared_result();
```

Comment:
Using something called `importer_adapter` as something other than an adapter seems like a smell, but that seems to be how we are using this elsewhere.

Ideally, we have a channel and we get the receivers from that before handing the sender to the adapter. I think this file needs to be cleaned up a bit anyway, so might just push that to a cleanup issue.

Reply:
Usually, the service defines its requirements to other services via ports. Importer is one of them. Usually, we pass the importer in the constructor of the service where it calls `get_steram_of_blocks` or something like that.

I don't think explicit work with the channel is better; in this case, a lot of implementation details are revealed. What if later we decide to use the `mpsc` channel or another channel? In this case, you need to update all places to use it. By using the importer and requiring that it provide some stream with blocks, we are more flexible and don't expose implementation details.